### PR TITLE
Feat: Add support for OpenGauss database types

### PR DESCRIPTION
**Pull request by lgj101**
### Description of your changes

Add support for OpenGauss database types

Fixes #

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary.
- [ ] Run `yarn lint` to ensure the frontend changes are ready for review.
- [ ] Run `make reviewable`to ensure the server changes are ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Review by chivalryq — COMMENTED** [review]
Thanks for contributing. Please fix the CI.

File: go.mod [hunk 1]
```diff
@@ -106,8 +106,9 @@ require (
 	github.com/grafana/grafana v1.9.2-0.20230216173926-a0bea04a0274
 	github.com/julienschmidt/httprouter v1.3.0
 	github.com/kubevela/workflow v0.6.0
-	github.com/oam-dev/kubevela v1.9.1
```
Comment (chivalryq) [inline go.mod:109]: kubevela v1.9.5 imports some API break changes. You can rollback to v1.9.4 if you don't want to import too much unrelated changes in this PR.

File: pkg/server/infrastructure/datastore/datastore_test.go [hunk 2]
```diff
@@ -31,6 +32,7 @@ import (
 	"go.mongodb.org/mongo-driver/mongo"
 	"go.mongodb.org/mongo-driver/mongo/options"
 	mysqlgorm "gorm.io/driver/mysql"
+	opengaussgorm "gorm.io/driver/postgres"
```
[the remark this answers is not in the document]
Comment (lgj101) [reply pkg/server/infrastructure/datastore/datastore_test.go:35]: ok let me modify it

File: go.mod (hunk 1 above)
Comment (lgj101) [reply go.mod:109]: Modified, please help me take a look again

File: .github/workflows/server-test.yml [hunk 3]
```diff
@@ -72,6 +72,16 @@ jobs:
           mysql database: 'kubevela'
           mysql root password: 'kubevelaSQL123'
 
+      - name: Set up OpenGauss
+        uses: lgj101/opengauss-action@v1.6
+        with:
+          GS_VERSION: 'latest'
```
Comment (chivalryq) [inline .github/workflows/server-test.yml:78]: Please specify a version in case the "latest" version make CI unstable.

File: pkg/server/infrastructure/datastore/opengauss/opengauss.go [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+	database string
+}
+
+// New opengauss datastore instance
+func New(ctx context.Context, cfg datastore.Config) (datastore.DataStore, error) {
+
+	db, err := gorm.Open(postgresorm.Open(cfg.URL), &gorm.Config{
+		NamingStrategy:                           sqlnamer.SQLNamer{},
+		DisableForeignKeyConstraintWhenMigrating: true,
+		Logger:                                   logger.Default.LogMode(logger.Silent),
```
Comment (chivalryq) [inline pkg/server/infrastructure/datastore/opengauss/opengauss.go:48]: Is it ok to set logger level to `Silent`?

File: .github/workflows/server-test.yml (hunk 3 above)
Comment (lgj101) [reply .github/workflows/server-test.yml]: Okay, modified

File: pkg/server/infrastructure/datastore/opengauss/opengauss.go (hunk 4 above)
Comment (lgj101) [reply pkg/server/infrastructure/datastore/opengauss/opengauss.go]: I changed it to an Error level

**Comment by NeerajGartia21** [comment]
Thanks for your contribution @lgj101 . This PR looks good to me !!
Since the whole implementation of OpenGauss and PostgreSQL are almost similar, I think it is better if we name the driver as PostgreSQL and in the Documentation we can add a note about using OpenGauss as PostgreSQL driver.

File: .github/workflows/server-test.yml [hunk 5]
```diff
@@ -72,6 +72,16 @@ jobs:
           mysql database: 'kubevela'
           mysql root password: 'kubevelaSQL123'
 
+      - name: Set up OpenGauss
+        uses: lgj101/opengauss-action@v1.6
```
Comment (chivalryq) [inline .github/workflows/server-test.yml:76]: Is this really setup a OpenGauss DB? I checked https://github.com/Harmon758/postgresql-action/compare/master...lgj101:opengauss-action:master and it seems to change the exposed variable only. If this is only changes I think we can just uses the origin action and implement the Postgres driver.

After all if we setup a Postgres here we can only checked the compatibility with Postgres. We can't say VelaUX "supports OpenGauss" because we don't have test. 

However if we implement the Postgres driver and we can add notes in document that because OpenGauss is compatible with Postgres, you can also use OpenGauss.

Comment (chivalryq) [reply .github/workflows/server-test.yml:76]: I'm sorry, yes this action setup a opengauss db. Given the popularity of postgres and the compatibility between them, we can just rename the `driver/opengauss` to `driver/postgresql` and keep the openguass test.

Comment (lgj101) [reply .github/workflows/server-test.yml:76]: Okay, I have already made the modifications like this

**Review by chivalryq — APPROVED** [review]
LGTM
